modules/rockyouBidAdapter.js

```diff
@@ -0,0 +1,357 @@
+import * as utils from 'src/utils';
+import { Renderer } from 'src/Renderer';
+import { BANNER, VIDEO } from 'src/mediaTypes';
+import { registerBidder } from 'src/adapters/bidderFactory';
+
+const BIDDER_CODE = 'rockyou';
+
+const BASE_REQUEST_PATH = 'https://tas.rockyou.net/servlet/rotator/';
+const IFRAME_SYNC_URL = 'https://prebid.tex-sync.rockyou.net/usersync2/tas';
+const VAST_PLAYER_LOCATION = 'https://rya-static.rockyou.com/rya/js/PreBidPlayer.js';
+export const ROTATION_ZONE = 'openrtbprod';
+
+let isBidRequestValid = (bid) => {
+  return !!bid.params && !!bid.params.placementId;
+};
+
+/**
+* The RockYou Ad Serving system currently only accepts one placementId
+* per Ad request. For this reason, the first placementId indicated
+* will be chosen as the predominant placementId for this request.
+*/
+let determineOptimalPlacementId = (bidRequest) => {
+  return bidRequest.params.placementId;
+}
+
+let determineOptimalRequestId = (bidRequest) => {
+  return bidRequest.bidId;
+}
+
+let buildSiteComponent = (bidRequest) => {
+  let topLocation = utils.getTopWindowLocation();
+
+  let site = {
+    domain: topLocation.hostname,
+    page: topLocation.href,
+    ref: topLocation.origin
+  };
+
+  return site;
+}
+
+let buildDeviceComponent = (bidRequest) => {
+  let device = {
+    js: 1,
+    language: ('language' in navigator) ? navigator.language : null
+  };
+
+  return device;
+};
+
+let extractValidSize = (bidRequest) => {
+  let width = null;
+  let height = null;
+
+  if (!utils.isEmpty(bidRequest.sizes)) {
+    // Currently only the first size is utilized
+    let firstSize = bidRequest.sizes[0];
+    width = firstSize[0];
```

Comment:
this won't work if sizes is defined like `[300,250]`. You'll need to check for that condition.

Reply:
just to clarify, sizes can be List[List[X,Y]] or List[X,Y] ?

Reply:
that's right. There's a helper function you can use: https://github.com/prebid/Prebid.js/blob/master/src/utils.js#L113

Sorry this is a pain. The core should handle this but it's hard to update because every adapter might be doing it differently.

Reply:
Not a problem. Is the "HxW" vs [H,W] format (as returned by the helper function) more universal?

Reply:
I think so. But the `[H,W]` | `[[H,W]]` format is what DFP uses so that's what our external API looks like. The core should normalize this so each adapter doesn't have to.